shopping-cart/app/shoppingcart/checkout/address/CheckoutAddressPageController.java

```diff
@@ -24,66 +22,66 @@
 import play.mvc.Call;
 import play.mvc.Result;
 import play.twirl.api.Html;
-import shoppingcart.common.CartController;
 import shoppingcart.common.StepWidgetBean;
+import shoppingcart.common.SunriseFrameworkCartController;
 
 import javax.annotation.Nullable;
 import javax.inject.Inject;
-import javax.inject.Singleton;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 import java.util.concurrent.CompletionStage;
 
-import static common.utils.FormUtils.extractAddress;
-import static common.utils.FormUtils.extractBooleanFormField;
 import static io.sphere.sdk.utils.FutureUtils.exceptionallyCompletedFuture;
 import static io.sphere.sdk.utils.FutureUtils.recoverWithAsync;
 import static java.util.concurrent.CompletableFuture.completedFuture;
 
-@Singleton
-public class CheckoutAddressPageController extends CartController {
+@HttpContextScoped
+public class CheckoutAddressPageController extends SunriseFrameworkCartController {
 
-    protected final Form<CheckoutShippingAddressFormData> shippingAddressUnboundForm;
-    protected final Form<CheckoutBillingAddressFormData> billingAddressUnboundForm;
+    @Inject
+    protected CheckoutAddressPageContentFactory checkoutAddressPageContentFactory;
+    protected Form<CheckoutShippingAddressFormData> shippingAddressUnboundForm;
+    protected Form<CheckoutBillingAddressFormData> billingAddressUnboundForm;
 
     @Inject
-    public CheckoutAddressPageController(final ControllerDependency controllerDependency, final ProductDataConfig productDataConfig,
-                                         final FormFactory formFactory) {
-        super(controllerDependency, productDataConfig);
+    public void initializeForms(final FormFactory formFactory) {
         this.shippingAddressUnboundForm = formFactory.form(CheckoutShippingAddressFormData.class);
         this.billingAddressUnboundForm = formFactory.form(CheckoutBillingAddressFormData.class);
     }
 
     @AddCSRFToken
     public CompletionStage<Result> show(final String languageTag) {
-        final UserContext userContext = userContext(languageTag);
-        return getOrCreateCart(userContext, session())
-                .thenApplyAsync(cart -> {
-                    final CheckoutAddressPageContent pageContent = createPageContent(cart, userContext);
-                    return ok(renderCheckoutAddressPage(cart, pageContent, userContext));
-                }, HttpExecution.defaultContext());
+        return getOrCreateCart()
+                .thenApplyAsync(this::showCheckoutAddressPage, HttpExecution.defaultContext());
     }
 
     @RequireCSRFCheck
+    @SuppressWarnings("unused")
     public CompletionStage<Result> process(final String languageTag) {
-        final UserContext userContext = userContext(languageTag);
+        return getOrCreateCart()
+                .thenComposeAsync(this::processAddressForm, HttpExecution.defaultContext());
+    }
+
+    private Result showCheckoutAddressPage(final Cart cart) {
```

Comment:
Just to make it clear, but I want to avoid is having different structures per controller. If you know how the structure of a controller looks like, you should be able to know all the other controllers, because all of them will have exactly the same structure. It's a way to be consistent. So for every change, we have to realize all the rest will be looking exactly the same.